pyfr/solvers/baseadvecdiff/system.py

```diff
@@ -15,16 +15,33 @@ def deps(dk, *names): return self._kdeps(k, dk, *names)
         g1 = self.backend.graph()
         g1.add_mpi_reqs(m['scal_fpts_recv'])
 
+        # Perform post-processing of the previous solution stage
+        g1.add_all(k['eles/filter_solution'])
+
         # Interpolate the solution to the flux points
-        g1.add_all(k['eles/disu'])
+        g1.add_all(k['eles/disu'], deps=k['eles/filter_solution'])
 
         # Pack and send these interpolated solutions to our neighbours
         g1.add_all(k['mpiint/scal_fpts_pack'], deps=k['eles/disu'])
         for send, pack in zip(m['scal_fpts_send'], k['mpiint/scal_fpts_pack']):
             g1.add_mpi_req(send, deps=[pack])
 
+        # If entropy filtering, compute entropy bounds
+        g1.add_mpi_reqs(m['ent_fpts_recv'])
+
+        # Pack and send the entropy values to neighbors
+        g1.add_all(k['mpiint/ent_fpts_pack'], deps=k['eles/filter_solution'])
+        for send, pack in zip(m['ent_fpts_send'], k['mpiint/ent_fpts_pack']):
+            g1.add_mpi_req(send, deps=[pack])
+
+        # Compute common entropy minima at internal/boundary interfaces
+        g1.add_all(k['iint/comm_entropy'],
+                   deps=k['eles/filter_solution'] + k['mpiint/ent_fpts_pack'])
+        g1.add_all(k['bcint/comm_entropy'],
+                   deps=k['eles/filter_solution'] + k['eles/disu'])
```

Comment:
Does this also need to wait for the packing, or is the packing a false dependency?

Reply:
Removed.

Reply:
So you probably want a false dependency somewhere on the packing kernels to ensure they get executed sooner rather than later.  Otherwise, it is worth just going through and making sure that no dependencies are missing, as this can result in really weird and intermittent bugs.

Reply:
So keep the pack dependency there? For the comm_flux kernels it's a hard dependency, but the comm_entropy kernels don't need it.

Reply:
Yes.  Although I think you can remove the filter solution dependency from the BC int kernels since they depend on disu which in turn depends on the filter kernel.

Reply:
Done